spec/ParseQuery.Aggregate.spec.js

```diff
@@ -0,0 +1,412 @@
+        done();
+      }).catch(done.fail);
+  });
+
+  it('group sum query', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        group: { objectId: null, total: { $sum: '$score' } },
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .then((resp) => {
+        expect(resp.results[0].hasOwnProperty('objectId')).toBe(true);
+        expect(resp.results[0].objectId).toBe(null);
+        expect(resp.results[0].total).toBe(50);
+        done();
+      }).catch(done.fail);
+  });
+
+  it('group count query', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        group: { objectId: null, total: { $sum: 1 } },
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .then((resp) => {
+        expect(resp.results[0].hasOwnProperty('objectId')).toBe(true);
+        expect(resp.results[0].objectId).toBe(null);
+        expect(resp.results[0].total).toBe(4);
+        done();
+      }).catch(done.fail);
+  });
+
+  it('group min query', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        group: { objectId: null, minScore: { $min: '$score' } },
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .then((resp) => {
+        expect(resp.results[0].hasOwnProperty('objectId')).toBe(true);
+        expect(resp.results[0].objectId).toBe(null);
+        expect(resp.results[0].minScore).toBe(10);
+        done();
+      }).catch(done.fail);
+  });
+
+  it('group max query', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        group: { objectId: null, maxScore: { $max: '$score' } },
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .then((resp) => {
+        expect(resp.results[0].hasOwnProperty('objectId')).toBe(true);
+        expect(resp.results[0].objectId).toBe(null);
+        expect(resp.results[0].maxScore).toBe(20);
+        done();
+      }).catch(done.fail);
+  });
+
+  it('group avg query', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        group: { objectId: null, avgScore: { $avg: '$score' } },
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .then((resp) => {
+        expect(resp.results[0].hasOwnProperty('objectId')).toBe(true);
+        expect(resp.results[0].objectId).toBe(null);
+        expect(resp.results[0].avgScore).toBe(12.5);
+        done();
+      }).catch(done.fail);
+  });
+
+  it('limit query', (done) => {
+    const options = Object.assign({}, masterKeyOptions, {
+      body: {
+        limit: 2,
+      }
+    });
+    rp.get(Parse.serverURL + '/aggregate/TestObject', options)
+      .then((resp) => {
+        expect(resp.results.length).toBe(2);
+        done();
+      }).catch(done.fail);
+  });
+
+  it('sort ascending query', (done) => {
```

Comment:
This looks like it duplicates functionality we already have in standard queries, is this needed for something in particular where the original won't do?

Reply:
As the user builds their aggregate queries they have the option to limit, sort, skip. This test is also there for Postgres support

Reply:
Yeah I realized that right after I noted this, you're good here.